README
Module-Starter-Plugin-CGIApp

This is a plugin for Module::Starter that builds you a skeleton 
CGI::Application module with all the extra files needed to package it for 
CPAN. You can customize the output using HTML::Template.

INSTALLATION

If you have the Module::Signature module installed, you can verify the
integrity of this distribution by typing:

    cpansign -v

Module::Signature requires GnuPG which you can get from: 
http://www.gnupg.org/download/
A binary version for Windows is available from that page.
        
Then to install this module, run the following commands:

    perl Build.pl
    ./Build
    ./Build test
    ./Build install
    
On Windows you don't need to type the preliminary ./

Alternatively, you can do:

    perl Makefile.PL
    make
    make test
    make install

If you are using ActiveState perl on Windows and you don't have a make 
utility, you can download nmake from:
http://download.microsoft.com/download/vc15/patch/1.52/w95/en-us/nmake15.exe
(Replace make with nmake in the instructions above if you are using it.)

If you are using Strawberry perl on Windows, it comes with a make utility 
called dmake.  (Replace make with dmake in the instructions above if you are
using it.)

NOTE: At the moment, some of the tests will fail on Windows if the path to the 
directory you unpacked the distribution in has spaces in it.  Either skip the 
tests or move to another directory.

DEPENDENCIES

This module requires these other modules:

perl 5.008
File::ShareDir
HTML::Template
Module::Starter

At build time you will need the above modules plus the following if you want
to run the tests:

File::Copy::Recursive
File::DirCompare
Test::More
Test::MockTime
Time::Piece            1.12

The signature can be tested by installing the following module:

Module::Signature

and setting the environment variable TEST_SIGNATURE to 1.

Further tests are enabled by setting the environment variable RELEASE_TESTING
to 1 and installing the following modules:

Perl::Critic           1.098
Perl::MinimumVersion   1.20
Pod::Coverage          0.18
Pod::Simple            3.07
Test::CPAN::Meta       0.12
Test::DistManifest     1.003
Test::HasVersion       0.012
Test::Kwalitee
Test::MinimumVersion   0.008
Test::Perl::Critic     1.01
Test::Pod              1.26
Test::Pod::Coverage    1.08

They are all available on CPAN (http://www.cpan.org)

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the perldoc 
command.

    perldoc Module::Starter::Plugin::CGIApp

You can also look for information at:

    Search CPAN
        http://search.cpan.org/dist/Module-Starter-Plugin-CGIApp

    CPAN Request Tracker:
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Module-Starter-Plugin-CGIApp

    AnnoCPAN, annotated CPAN documentation:
        http://annocpan.org/dist/Module-Starter-Plugin-CGIApp

    CPAN Ratings:
        http://cpanratings.perl.org/d/Module-Starter-Plugin-CGIApp

COPYRIGHT AND LICENCE

Copyright (C) 2015, Consolidated Braincells Inc.  All Rights Reserved.

This distribution is free software; you can redistribute it and/or modify it
under the terms of either:

a) the GNU General Public License as published by the Free Software
Foundation; either version 1, or (at your option) any later version, or

b) the Artistic License version 1.0 or a later version.

The full text of the license can be found in the LICENSE file included
with this distribution.


Changes
Revision history for Module-Starter-Plugin-CGIApp


0.43    Sun Sep 20 04:43:35 EDT 2015

        Internal:
        ---------

        Fixed to work with newer versions of Module-Starter. RT #101984, #66240

        Added dependency on Module::Starter >= 1.70

        In Created distributions:
        -------------------------

        t/boilerplate.t is added to MANIFEST.

        More work still needs to be done for Module-Starter parity...

0.42	Wed May 16 19:17:15 EDT 2012

        Internal:
        ---------

        The license of this module is "Same as Perl" which somewhere along
        the line I thought meant GPL 2+/Artistic 2+ when actually it is
        GPL 1+/Artistic 1+.  This is now fixed in all the relevant places.

        In Created distributions:
        -------------------------

        The same license confusion occurred here too.  Fixed.

0.41    Thu May 10 23:54:01 EDT 2012

        Internal:
        ---------

        Forgot to mention File::ShareDir as a dependency.

        While I'm at it, I normalized the format of copyright statements
        everywhere and updated the copyright year of this distro to 2012.

0.40    Thu May 10 22:58:23 EDT 2012

        Internal:
        ---------

        Breakage caused by newer versions of Module::Starter (RT#71734) has
        been fixed.

        Module has been updated to use the latest features of Module::Starter.

        Author tests have been moved into xt directory and brought up to
        current best practices.

        Templates have been moved into a share directory using File::ShareDir.

        Templates have been improved in various minor ways.

        In Created distributions:
        -------------------------

        Author tests are brought up to current best practices and now go into
        the xt directory so they don't have to be run by end users.

        Templates go into a shared directory using File::ShareDir instead
        of the old method of putting them in @INC

        Instead of pgp.mit.edu, t/00-signature.t will now use
        pool.sks-keyservers.net for key queries.

        MANIFEST.SKIP will also skip .git

0.30    Sat Mar  6 22:54:25 EST 2010
        cgiapp-starter and titanium-starter had got broken during the last
        release due to an errant local and were unable to find the default
        template directory. They should work correctly now.

        t/boilerplate.t in generated dists had got broken during the last
        set of updates.  It should work correctly now.  It will now also
        check that you do not have a boilerplate abstract in your buildscript.

        Previously there was support for a generated dist with multiple
        modules but it didn't really work.  Now it does.  This justifies
        a bump in the version number I think.

        The RT address in generated modules (for bug reporting) is based on
        the distro name instead of the module name now.

0.22    Thu Mar  4 01:52:09 EST 2010
        t/00-signature.t was wrong in the module and generated dists.
        Now it will only be carried out if the envvar TEST_SIGNATURE is set.

        Unneeded dependencies deleted from the build scripts in the module
        and generated dists.

        README in the module and generated dists updated to reflect these
        changes.

0.21    Wed Mar  3 15:28:02 EST 2010
        Fixed test failures on Windows (and possibly other platforms with
        disgusting perverted non-unixy line endings.)  Thanks csjewell,
        jdb, and samuraijack on #win32@irc.perl.org and almut, ambrus, and
        banesong on Perl Monks for their help in solving these problems.
        
        Upped the recommended perl version to 5.008.

0.20    Sun Feb 28 14:02:44 EST 2010
        t/test-app.t in generated dists now passes an object to $mech->app
        to match what is done is server.pl
        
        Tests in generated dists now follow Test::XT recommendations.
        
        If the license of a generated dist is 'perl', the license blurb in
        the module and README explicitly mentions GPL+Artistic instead of
        just 'same as perl'.  A LICENSE file is added to the dist where you
        can put the license terms.

        Hopefully the problem with Build mistakenly trying to run the 
        Makefile.PLs from templates is resolved.

        Revamped test suite should be more comprehensive than before.  In
        line with Test::XT recommendations, test relevant to the author
        only, will be skipped unless envvar RELEASE_TESTING=1.  See README.

0.10    Thu Jan 15 21:12:55 EST 2009
        Added a placeholder file in t/www in generated distributions as the
        empty directory was causing error messages.
        
        More build fixes for Windows.  I think I actually have got it right
        this time.  One problem remains.  Tests will fail if the directory 
        path to where you have unpacked the module has spaces anywhere in it.
        What's up with that?
        
        Ensured that MANIFEST in generated distributions always uses / as the
        directory seperator regardless of operating system.

        Added some entries to MANIFEST.SKIP in generated distributions to
        stop the signature test from failing under Windows.
        
        Thanks to Ron Savage and Cosimo Streppone from the cgiapp list and
        ikegami, cdarke, and andreas1234567 on Perl Monks for their help in 
        solving Windows issues.
        
0.08    Fri Jan  9 18:24:07 EST 2009
        This version just has changes to the build process which hopefully
        will make it work better under Windows.

0.07    Mon Jan  5 17:24:30 EST 2009
        No longer build depends on Test::Builder.
        
        By popular demand, templates use $self instead of $c.
        
        Distributions now include a directory, t/www for static files.
        server.pl has been updated to use it.
        

Build.PL
use strict;
use warnings;
use File::Find;
use Module::Build;

my $builder = Module::Build->new(
    module_name         => 'Module::Starter::Plugin::CGIApp',
    license             => 'perl',
    dist_author         => 'Jaldhar H. Vyas <jaldhar@braincells.com>',
    dist_version        => '0.43',
    share_dir           => 'share',
    pm_files            => {
        'lib/Module/Starter/Plugin/CGIApp.pm' => 'lib/Module/Starter/Plugin/CGIApp.pm',
    },
    script_files            => {
        'script/cgiapp-starter' => [],
        'script/titanium-starter' => [],
    },
    build_requires      => {
        'File::Copy::Recursive' => 0,
        'File::DirCompare'      => 0,
        'Test::More'            => 0,
        'Test::MockTime'        => 0,
        'Time::Piece'           => 1.12,
    },
    requires            => {
        'perl'                  => 5.008,
        'File::ShareDir'        => 0,
        'Module::Starter'       => 1.70,
        'HTML::Template'        => 0,
    },
    meta_merge => {
        resources => {
            repository  => 'http://github.com/jaldhar/Module-Starter-Plugin-CGIApp',
            homepage    => 'http://jaldhar.github.com/Module-Starter-Plugin-CGIApp/',
        },
    },
    add_to_cleanup      => [ 'Module-Starter-Plugin-CGIApp-*' ],
    create_makefile_pl  => 'small',
    sign                => 1,
);

$builder->create_build_script();

